syntax = "proto3";

service Iterator {
    rpc GetThroughput(GetThroughputRequest) returns (GetThroughputResponse) {}
    rpc NotifySaveCheckpoint(NotifySaveCheckpointRequest) returns (NotifySaveCheckpointResponse) {}
}

message GetThroughputRequest {
}

message GetThroughputResponse {
    bool success = 1;
    bool ready = 2;
    float throughput = 3;
}

message NotifySaveCheckpointRequest {
}

message NotifySaveCheckpointResponse {
    bool success = 1;
}